include/sys/onoff.h

```diff
@@ -0,0 +1,533 @@
+ * result.
+ *
+ * This function is isr-capable, and may suspend unless client
+ * notification specifies no-wait.
+ *
+ * @param srv the service that will be used.
+ *
+ * @param cli a non-null pointer to client state providing
+ * instructions on synchronous expectations and how to notify the
+ * client when the request completes.  Behavior is undefined if client
+ * passes a pointer object associated with an incomplete service
+ * operation.
+ *
+ * @retval Non-negative on successful (initiation of) request
+ * @retval -EIO if service has recorded an an error
+ * @retval -EINVAL if the parameters are invalid
+ * @retval -EAGAIN if the reference count would overflow
+ * @retval -EWOULDBLOCK if a non-blocking request was made and could
+ *         not be satisfied
+ */
+__syscall int onoff_request(struct onoff_service *srv,
+			    struct onoff_client *cli);
+
+/**
+ * @brief Release a reserved use of an on-off service.
+ *
+ * The return value indicates the success or failure of an attempt to
+ * initiate an operation to release of the resource.  If initiation of
+ * the operation succeeds the result of result of the release
+ * operation itself is provided through the configured client
+ * notification method, possibly before this call returns.
+ *
+ * Note that the call to this function may succeed in a case where the
+ * actual release fails.  Always check the operation completion
+ * result.
+ *
+ * This function is isr-capable, and may suspend unless client
+ * notification specifies no-wait.
+ *
+ * @param srv the service that will be used.
+ *
+ * @param cli a non-null pointer to client state providing
+ * instructions on how to notify the client when release completes.
+ * Behavior is undefined if cli references an object associated with
+ * an incomplete service operation.
+ *
+ * @retval Non-negative on successful (initiation of) release
+ * @retval -EINVAL if the parameters are invalid
+ * @retval -EIO if service has recorded an an error
+ * @retval -EWOULDBLOCK if a non-blocking request was made and
+ *         could not be satisfied without potentially blocking.
+ * @retval -EALREADY if the service is already off or transitioning
+ *         to off
+ * @retval -EBUSY if the service is transitioning to on
+ */
+__syscall int onoff_release(struct onoff_service *srv,
+			    struct onoff_client *cli);
+/**
+ * @internal
+ */
+static inline bool z_impl_onoff_service_has_error(const struct onoff_service *srv)
+{
+	return (srv->flags & ONOFF_SERVICE_HAS_ERROR) != 0;
+}
+
+/**
+ * @brief Test whether an on-off service has recorded an error.
+ *
+ * This function can be used to determine whether the service has
+ * recorded an error.  Errors may be cleared by invoking
+ * onoff_service_reset().
+ *
+ * @return true if and only if the service has an uncleared error.
+ */
+__syscall bool onoff_service_has_error(const struct onoff_service *srv);
+
+/**
+ * @brief Clear errors on an on-off service and reset it to its off
```

Comment:
It doesn't.  They get queued just like any other transition request, and the client will be notified when the reset is complete.  If a client tries to reset after complete the call will fail, because reset is rejected if the service is not in an error state.  That's subtly indicated by one of the `EINVAL` retval descriptions but should be more clear: I've queued some rewording.